### PR TITLE
Upgrade kube dependency to 0.34.0

### LINKED ISSUE #529: Upgrade kube dependency to support GKE & macOS
**Is your feature request related to a problem? Please describe.**
`make run` fails on macOS with a GKE cluster:
```
make run
RUST_LOG="rudr=debug,kube=info" RUST_BACKTRACE=short cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.32s
     Running `target/debug/rudr`
[2020-01-30T13:39:37Z INFO  rudr] starting server
[2020-01-30T13:39:37Z INFO  rudr] apiserver:https://34.65.118.77
[2020-01-30T13:39:37Z ERROR rudr] Error prechecking CRDs applicationconfigurations: Error { inner: Error(Hyper(Error(Connect, Custom { kind: Other, error: Error { code: -67843, message: "The certificate was not trusted." } })), "https://34.65.118.77/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/applicationconfigurations.core.oam.dev")

```

**Describe the solution you'd like**
`make run` should work ok on macOS with a GKE cluster.

The easiest way is to upgrade kube to pick up https://github.com/clux/kube-rs/commit/2b50f0132c76ae2f8ea37ea7188340c577a4b61c.

**Describe alternatives you've considered**
I could use another cluster type, but I'm more comfortable with GKE. Also, this project will eventually need to upgrade kube, so it's only a matter of time.

**Additional context**
See https://github.com/technosophos/rust-k8s-controller/pull/3 for a similar upgrade.

### PR DESCRIPTION
**Pull request by messense**
`cargo build` & `cargo test` passes on my local macOS.

Fixes #529

### CODE REVIEW

**Comment by messense** [comment]
The ` Kubernetes E2E` test output isn't very helpful.

> Error: timed out waiting for the condition
> make: *** [kind-e2e] Error 1
> Makefile:37: recipe for target 'kind-e2e' failed

**Comment by wonderflow** [comment]
Yeah, it's currently hard to find out the reason. In my experience, this usually means the pod of rudr can't be running. Can you try to build the image , deploy and fix it locally

**Comment by messense** [comment]
> Yeah, it's currently hard to find out the reason. In my experience, this usually means the pod of rudr can't be running. Can you try to build the image , deploy and fix it locally.

To be clear, this is just a drive-by PR.
I am not very family with k8s, and I tried but failed to start k8s with Docker Desktop on macOS.

**Comment by wonderflow** [comment]
I tried your PR locally, but it fails on compiling:

```
  Compiling bytes v0.5.4
   Compiling futures-timer v3.0.1
   Compiling proc-macro2 v1.0.8
   Compiling libc v0.2.66
   Compiling syn v1.0.14
   Compiling indexmap v1.3.2
   Compiling num-traits v0.2.11
   Compiling num-integer v0.1.42
   Compiling num-iter v0.1.40
   Compiling memchr v2.3.2
   Compiling openssl-sys v0.9.54
   Compiling unicase v2.6.0
   Compiling backtrace-sys v0.1.32
   Compiling proc-macro-nested v0.1.3
   Compiling futures-channel v0.3.4
   Compiling unicode-normalization v0.1.12
   Compiling thread_local v1.0.1
   Compiling miniz_oxide v0.3.6
   Compiling serde v1.0.104
   Compiling rustversion v1.0.2
   Compiling encoding_rs v0.8.22
   Compiling k8s-openapi v0.7.1
   Compiling textwrap v0.11.0
   Compiling humantime v1.3.0
   Compiling want v0.3.0
error: `core::slice::<impl [T]>::len` is not yet stable as a const fn
   --> /home/jianbo/.cargo/registry/src/github.com-1ecc6299db9ec823/bytes-0.5.4/src/bytes.rs:130:18
    |
130 |             len: bytes.len(),
    |                  ^^^^^^^^^^^

   Compiling aho-corasick v0.7.8
error: aborting due to previous error

error: Could not compile `bytes`.
warning: build failed, waiting for other jobs to finish...
```

**Comment by messense** [comment]
@wonderflow It requires rustc 1.41(current stable version).

**Comment by wonderflow** [comment]
Thanks @messense, I have successfully build and it crashes as the following errors:

```
root@leijuan-thinkstation-p720:/home/jianbo/rudr# kubectl logs -f rudr-764f4c5d8d-tn7sh
2020-02-17 02:56:11 INFO [rudr:src/main.rs:73] starting server
2020-02-17 02:56:11 INFO [rudr:src/main.rs:33] Loading in-cluster config
2020-02-17 02:56:11 INFO [rudr:src/main.rs:77] apiserver:https://10.96.0.1:443
2020-02-17 02:56:11 INFO [kube::api::reflector:/home/jianbo/.cargo/registry/src/github.com-1ecc6299db9ec823/kube-0.25.0/src/api/reflector.rs:107] Starting Reflector for RawApi { resource: "componentschematics", group: "core.oam.dev", namespace: Some("default"), version: "v1alpha1", prefix: "apis" }
2020-02-17 02:56:11 INFO [rudr:src/main.rs:173] Watcher is running
2020-02-17 02:56:11 INFO [rudr:src/main.rs:33] Loading in-cluster config
2020-02-17 02:56:11 INFO [rudr:src/main.rs:177] Health server is running on 0.0.0.0:8080
thread '<unnamed>' panicked at 'there is no reactor running, must be called from the context of Tokio runtime', src/libcore/option.rs:1188:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Any', src/libcore/result.rs:1188:5
```

**Comment by messense** [comment]
@wonderflow Great, I'll look into it later.

**Comment by wonderflow** [comment]
You could rebase the master branch and the CI will give more detail logs

**Comment by messense** [comment]
@wonderflow Still can't spot the problem in the detailed logs.

File: src/main.rs [hunk 1]
```diff
@@ -23,21 +25,22 @@ use rudr::schematic::{
 
 const DEFAULT_NAMESPACE: &str = "default";
 
-fn kubeconfig() -> kube::Result<kube::config::Configuration> {
+async fn kubeconfig() -> kube::Result<kube::config::Configuration> {
     // If env var is set, use in cluster config
     match std::env::var("KUBERNETES_PORT") {
         Ok(_val) => {
             info!("Loading in-cluster config");
```
Comment (wonderflow) [inline src/main.rs]: It seems it's blocked by some reason and always printing `Loading in-cluster config`.

**Comment by wonderflow** [comment]
Thanks so much for this great PR, but it's better if you could setup test environment.

I'm not very sure that will it possible that the `health_server` won't working?

File: src/main.rs (hunk 1 above)
Comment (messense) [reply src/main.rs]: This log is from `sync_status`

**Comment by messense** [comment]
`Kubernetes E2E` test passes.

File: src/schematic/scopes.rs [hunk 2]
```diff
@@ -45,38 +45,38 @@ impl OAMScope {
         }
     }
     /// create will create a real scope instance
-    pub fn create(&self, owner: meta::OwnerReference) -> Result<(), Error> {
+    pub async fn create(&self, owner: meta::OwnerReference) -> Result<(), Error> {
```
[the remark this answers is not in the document]
Comment (messense) [reply src/schematic/scopes.rs]: `kube-rs` needs `async`/`await`, and async is invasive, which means that async fns needs to be called in async fns.

**Comment by messense** [comment]
To upgrade `kube-rs`, we also need to upgrade `reqwest` to  0.10 which depends on `hyper` 0.13.

See https://github.com/dtolnay/async-trait for `#[async_trait]` usage.

File: healthscope/src/main.rs [hunk 3]
```diff
@@ -70,11 +66,12 @@ fn main() -> Result<(), Error> {
         let client = APIClient::new(cfg_watch);
         let mut cnt = 0;
         loop {
-            let req = healthscope_resource.list(&ListParams::default())?;
-            match client.request::<ObjectList<HealthScopeObject>>(req) {
+            let req = healthscope_resource.list(&ListParams::default()).unwrap();
```
[the remark this answers is not in the document]
Comment (messense) [reply healthscope/src/main.rs]: I'll let you review the over all approach before addressing these small nits.

**Pull request closed without merging**